Assets/00_Content/Scripts/Vikings/Viking.cs

```diff
@@ -89,6 +121,31 @@ public class Viking : Interactable {
 			ChangeState(new BrawlingVikingState(this, CurrentChair.Table));
 		}
 
+		public void MakeSpinAttack() {
+			if (!IsAttacking) {
+				IsAttacking = true;
+				StartCoroutine(SpinAttack());
+				Invoke(nameof(FinishAttack), vikingData.spinAttackDuration);
+			}
+		}
+
+		private IEnumerator SpinAttack() {
+			while(IsAttacking) {
+				yield return null;
+				transform.Rotate(Vector3.up, vikingData.spinAttackSpeed * Time.deltaTime);
+			}
+		}
+
+		private void FinishAttack() {
+			StopAllCoroutines();
+			IsAttacking = false;
+			isAttacked = false;
+
+			foreach (MeshRenderer hitHighlightMeshRenderer in hitHighlightMeshRenderers) {
+				hitHighlightMeshRenderer.material = normalMaterial;
+			}
+		}
```

Comment:
Could these attack functions be moved/refactored to a state as they are very state dependent?

Reply:
I will move most of the stuff out from PlayerComponent into a new PlayerBrawling